crates/typst-eval/src/code.rs

```diff
@@ -54,7 +54,17 @@ fn eval_code<'a>(
 
         output = ops::join(output, value).at(span)?;
 
-        if vm.flow.is_some() {
+        if let Some(event) = &vm.flow {
+            if matches!(output, Value::Content(_)) {
+                if let FlowEvent::Return(span, Some(_), false) = event {
+                    vm.engine.sink.warn(warning!(
+                        *span,
+                        "unconditional explicit return value discards content";
```

Comment:
how about "this return unconditionally discards the content before it"

Reply:
yeah, that seems more readable